### PR TITLE
Prevent middleware error on older Django versions (<1.8)

### PR DESCRIPTION
**Pull request by wsabransky**

### CODE REVIEW

**Comment by reyang** [comment]
@zoidbergwill FYI

**Comment by reyang** [comment]
@wsabransky does this mean we can support much lower version at https://github.com/census-instrumentation/opencensus-python/blob/master/contrib/opencensus-ext-django/setup.py#L42?

**Comment by wsabransky** [comment]
@reyang Yep, I've tested this on 1.6.5 and it runs.

**Comment by wsabransky** [comment]
Hmm, it seems like unit tests are failing for reasons potentially not related to this PR. Maybe due to this being moved into its own package?

**Comment by reyang** [comment]
@wsabransky Great! Probably we can change the required Django version in `setup.py`. What would be your suggestion on the lowest version?

**Comment by reyang** [comment]
The CI failure was introduced by the PostgreSQL lib update.
I have a fix in #602.

**Comment by wsabransky** [comment]
@reyang I don't know enough about Django before 1.6.x to say with confidence that it would work. But 1.6.x is definitely safe.

**Comment by reyang** [comment]
@wsabransky Great to know. Thank you!

**Comment by reyang** [comment]
@wsabransky I have the CI fixed, would you rebase to the latest master and try?

**Comment by wsabransky** [comment]
@reyang done

File: contrib/opencensus-ext-django/opencensus/ext/django/middleware.py [hunk 1]
```diff
@@ -99,7 +99,11 @@ def _set_django_attributes(span, request):
         return
 
     user_id = django_user.pk
-    user_name = django_user.get_username()
+    try:
+        user_name = django_user.get_username()
+    except Exception:
```
Comment (zoidbergwill) [inline contrib/opencensus-ext-django/opencensus/ext/django/middleware.py]: I think it'd be better checking the Django version or catching a more specific error then a general exception?

**Comment by zoidbergwill** [comment]
@reyang Thanks for the ping, when we add support for testing multiple Django versions, we can support all the currently maintained versions then probably, which is great.

**Comment by c24t** [comment]
FWIW we've been setting the minimum django version according to [django's support timeline](https://docs.djangoproject.com/en/2.2/internals/release-process/#supported-versions) even though the opencensus middleware is technically compatible with older versions. We do the same thing for flask.

Since we only depend on django to instrument it (and aren't e.g. running a django service ourselves), most of the usual security concerns don't apply. On the other hand, it makes our lives easier if we always set the minimum required versions of our dependencies such that it's impossible to accidentally install an old unsupported version.

File: contrib/opencensus-ext-django/opencensus/ext/django/middleware.py (hunk 1 above)
Comment (wsabransky) [reply contrib/opencensus-ext-django/opencensus/ext/django/middleware.py]: Good point. Changed this

**Comment by wsabransky** [comment]
@c24t Makes sense. I've reverted the supported minimum version change. Anything else you'd like to see to get this merged?

**Comment by c24t** [comment]
Looks like lint and coverage checks are still failing.

**Comment by wsabransky** [comment]
@c24t Any chance we could get this merged into a minor version update for the django extension?

**Comment by c24t** [comment]
@wsabransky in case you haven't seen it already: this was released in https://github.com/census-instrumentation/opencensus-python/releases/tag/v0.5.0.

**Pull request merged**